client/kubernetes_client_provider.go

```diff
@@ -13,51 +13,106 @@
 package client
 
 import (
+	"errors"
+	"strconv"
+	"os"
+	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
-// Provider for creation new kubernetes api client.
-type KubernetesClientProvider struct {
+var (
+	// UseUserToken - flag to enable/disable using user token to have access to k8s api.
+	UseUserToken bool
+)
+
+const (
+	// UserTokenAttr attribute name.
+	UserTokenAttr = "userToken"
+	// UseUserTokenEnvName env variable, boolean flag to enable/disable using user token to have access to k8s api.
+	UseUserTokenEnvName = "USE_USER_TOKEN"
+)
+
+func init() {
+	useUserTokenEnvValue, isFound := os.LookupEnv(UseUserTokenEnvName)
```

Comment:
I like the way how we organized config with deprecated bootstrapper, exec agent where we are able to configure an application with flag or environment variable:
https://github.com/eclipse/che/blob/6.19.6/agents/go-agents/exec-agent/main.go#L195
https://github.com/eclipse/che/blob/6.19.6/agents/go-agents/bootstrapper/cfg/cfg.go#L87

WDYT about introducing the similar config here?